bson.c

```diff
@@ -1212,7 +1260,7 @@ char* bson_to_zval(char *buf, HashTable *result, mongo_bson_conversion_options *
 
 				/* put it all together */
 				array_init(value);
-				add_assoc_stringl(value, "$ref", ns, ns_len-1, 1);
+				add_assoc_zval(value, "$ref", zns);
```

Comment:
Because I wanted to ensure `const char *buf` for this function, we couldn't use a `char *ns`, which is required by [`add_assoc_stringl_ex()`](http://lxr.php.net/xref/PHP_5_3/Zend/zend_API.c#1183) for reasons unknown. We can work around it by creating the string zval ourselves. We do this elsewhere in the driver (e.g. collection name property) and this is what `add_assoc_stringl_ex()` does internally anyway, so it should be fine.